timber-lint/src/main/java/timber/lint/WrongTimberUsageDetector.kt

```diff
@@ -122,8 +125,9 @@ class WrongTimberUsageDetector : Detector(), UastScanner {
       }
       if (current.isMethodCall()) {
         val psiMethod = (current as UCallExpression).resolve()
-        if (Pattern.matches(TIMBER_TREE_LOG_METHOD_REGEXP, psiMethod!!.name)
-          && context.evaluator.isMemberInClass(psiMethod, "timber.log.Timber")
+        if (psiMethod != null &&
```

Comment:
cuz `arrayOf` is part of the Kotlin stdlib and not resolvable

Reply:
It's an intrinsic that resolves to a regular Java array creation. That being said I don't understand the comment in the first place.

Reply:
My understanding is that Lint reference resolution is limited to types for which it can find the source.  Since we don't own the source of arrayOf (or any method that might wrap in this case), psiMethod will be null.

Reply:
Note quite -- it just needs to be on the classpath (which is why the earlier string format comparison resolves to a method in the special class kotlin.text.StringsKt__StringsJVMKt). Lint normally tries to put the android.jar classes into the classpath (if it's an android project), as well as the Kotlin runtime jars (if the unit test contains Kotlin).

Reply:
Dug into this a little more and it looks like arrayOf is on the classpath:

in `kotlin/text/StringsJVM.kt`
```
@kotlin.internal.InlineOnly
public inline fun String.Companion.format(format: String, vararg args: Any?): String = java.lang.String.format(format, *args)
```

in `kotlin/Library.kt`
```
public inline fun <reified @PureReifiable T> arrayOf(vararg elements: T): Array<T>
```

However, that resolution is returning null, because the KotlinUFunctionCallExpression.resolvedCall.resultingDescriptor (wat!) is a DeserializedCallableMemberDescriptor whose containingDeclaration is neither a LazyJavaPackageFragment or a DeserializedClassDescriptor, but rather of type org.jetbrains.kotlin.serialization.deserialization.builtins.BuiltInsPackageFragmentImpl.

See here: https://cs.android.com/android-studio/intellij-kotlin/+/master:uast/uast-kotlin/src/org/jetbrains/uast/kotlin/internal/kotlinInternalUastUtils.kt;l=389

Is this a UAST bug, @tnorbye?